# NGS Sequencing Technology and File Formats
https://learn.gencore.bio.nyu.edu/ngs-file-formats/
$~$

![alt text](image.png)

### Definitions

**read**: a single sequence produced from a sequencer. Think: a sequencing machine read a molecule and this is what it thinks it is.

**library**: a collection of DNA fragments that have been prepared for sequencing. This is generally talking about individual samples.

**flowcell**: a chip on which DNA is loaded and provided to the sequencer.

**lane**: one portion of a flowcell. Usually used for technical replicates or different samples.

**run**: an entire sequencing reaction from start to finish.

---

### How Sequencing Works
There are three main steps in NGS:
1. Sample collection/preparation
2. Amplification
3. Basecalling

##### Sample Collection and Preparation
![alt text](image-1.png)

##### Single-end vs Paired-end Sequencing
![alt text](image-2.png)
![alt text](image-3.png)

##### Library Prep - Nucleic Acids Isolation and Amplification
![alt text](image-4.png)

___

### Sequencing
![alt text](image-5.png)
https://youtu.be/fCd6B5HRaZ8

---
### FastA Format
FastA contains:
- Sequence name
- Description of the sequence (metadata, sequencer info, annotations, etc.)
  - The sequence itselfit can be either nucleic acids or amino acids as long as it adheres to the format.

Each sequence consists of at least two lines
   - the sequence header/identifier, starts with a ‘>’ to the first whitespace.
```
>Chr1 CHROMOSOME dumped from ADB: Jun/20/09 14:53; last updated: 2009-02-02
CCCTAAACCCTAAACCCTAAACCCTAAACCTCTGAATCCTTAATCCCTAAATCCCTAAATCTTTAAATCCTACATCCAT
GAATCCCTAAATACCTAATTCCCTAAACCCGAAACCGGTTTCTCTGGTTGAAAATCATTGTGTATATAATGATAATTTT
ATCGTTTTTATGTAATTGCTTATTGTTGTGTGTAGATTTTTTAAAAATATCATTTGAGGTCAATACAAATCCTATTTCT
TGTGGTTTTCTTTCCTTCACTTAGCTATGGATGGTTTATCTTCATTTGTTATATTGGATACAAGCTTTGCTACGATCTA
CATTTGGGAATGTGAGTCTCTTATTGTAACCTTAGGGTTGGTTTATCTCAAGAATCTTATTAATTGTTTGGACTGTTTA
TGTTTGGACATTTATTGTCATTCTTACTCCTTTGTGGAAATGTTTGTTCTATCAATTTATCTTTTGTGGGAAAATTATT
TAGTTGTAGGGATGAAGTCTTTCTTCGTTGTTGTTACGCTTGTCATCTCATCTCTCAATGATATGGGATGGTCCTTTAG
CATTTATTCTGAAGTTCTTCTGCTTGATGATTTTATCCTTAGCCAAAAGGATTGGTGGTTTGAAGACACATCATATCAA
AAAAGCTATCGCCTCGACGATGCTCTATTTCTATCCTTGTAGCACACATTTTGGCACTCAAAAAAGTATTTTTAGATGT
TTGTTTTGCTTCTTTGAAGTAGTTTCTCTTTGCAAAATTCCTCTTTTTTTAGAGTGATTTGGATGATTCAAGACTTCTC
GGTACTGCAAAGTTCTTCCGCCTGATTAATTATCCATTTTACCTTTGTCGTAGATATTAGGTAATCTGTAAGTCAACTC
ATATACAACTCATAATTTAAAATAAAATTATGATCGACACACGTTTACACATAAAATCTGTAAATCAACTCATATACCC
GTTATTCCCACAATCATATGCTTTCTAAAAGCAAAAGTATATGTCAACAATTGGTTATAAATTATTAGAAGTTTTCCAC
TTATGACTTAAGAACTTGTGAAGCAGAAAGTGGCAACACCCCCCACCTCCCCCCCCCCCCCCCACCCCCCAAATTGAGA
AGTCAATTTTATATAATTTAATCAAATAAATAAGTTTATGGTTAAGAGTTTTTTACTCTCTTTATTTTTCTTTTTCTTT
```
---

### FastQ Format

The format is similar to FastA, with differences in syntax as well as integration of quality scores. Each sequence requires at least 4 lines:
$~~~$ 1. The sequence header starts with an "@"
$~~~$ 2. The second line is the sequence
$~~~$ 3. The third line starts with "+"
$~~~$ 4. The fourth line are the quality scores

![1771530223110](image/0.NGSSequencingTechnologyandFileFormats/1771530223110.png)
![alt text](<Screenshot 2026-02-19 at 13.49.19.png>)

##### Quality Score
- A way to assign confidence to a particular base within a read
- Represents the probability of an incorrect basecall at that position.
- Common used to filter bases or entire reads under quality threshold
  - e.g. A read's mean quality score
  ###### Phred Quality Score (Q-score)
  - Numbers (0-40) in ASCII
  - Phred-64 was only ever used by Illumina
  - industry standard of Q30 (99.9% accuracy)
  - Error Probability: $$ P=10^\frac{-Q}{10} $$

---
### SAM Format
- It is the most basic, human readable format of the three
- It consists of a header, a row for every read in dataset, and 11 tab-delimited fields describing that read
  ###### Field Descriptions - 11 mandatory fields
  e.g. 
    - Bitwise Flag - explain certain features about the particular read
    - MapQ - how well the read aligned to the reference
    - CIGAR String - explain which part of the sequence aligns, which doesn’t, and everything in between. 
---
### BAM Format
Same format of SAM, except that it encoded in binary.

### CRAM Format
Similar to BAM as it retains the same information as SAM and is compressed.

---